assets/src/edit-story/components/canvas/gridview/useGridViewKeys.js

```diff
@@ -0,0 +1,278 @@
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useMemo, useState } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useKeyDownEffect } from '../../keyboard';
+import { useStory } from '../../../app';
+
+function useGridViewKeys(ref, gridRef, pageRefs, isRTL) {
+  const { arrangePage, currentPageId, pages } = useStory(
+    ({ state: { pages, currentPageId }, actions: { arrangePage } }) => ({
+      arrangePage,
+      currentPageId,
+      pages,
+    })
+  );
+
+  const [focusedPageId, setFocusedPageId] = useState(currentPageId);
+
+  const pageIds = useMemo(() => pages.map(({ id }) => id), [pages]);
+
+  // Navigate focus left, right, up, down
+  useKeyDownEffect(
+    ref,
+    { key: ['up', 'down', 'left', 'right'] },
+    ({ key }) => {
+      switch (key) {
+        case 'ArrowLeft':
+        case 'ArrowRight': {
+          const dir = getArrowDir(key, 'ArrowRight', 'ArrowLeft', isRTL);
+
+          if (dir === 0) {
+            return;
+          }
+
+          const index = pageIds.indexOf(focusedPageId);
+          const nextIndex = index + dir;
+
+          if (nextIndex < 0 || nextIndex === pageIds.length) {
+            return;
+          }
+
+          const pageId = pageIds[nextIndex];
+
+          setFocusedPageId(pageId);
+
+          const page = pageRefs.current && pageRefs.current[pageId];
+
+          focusOnPage(page);
+
+          break;
+        }
+        case 'ArrowUp':
+        case 'ArrowDown': {
+          const {
+            rows: numRows,
+            columns: numColumns,
+          } = getGridColumnAndRowCount(gridRef.current);
+          const currentIndex = pageIds.indexOf(focusedPageId);
+          const dir = key === 'ArrowDown' ? 1 : -1;
+
+          const currentRow = getRow(currentIndex, numColumns);
+          const currentColumn = getColumn(currentIndex, numColumns);
+
+          const nextIndex = getIndex(
+            currentRow + dir,
+            currentColumn,
+            numRows,
+            numColumns,
+            pageIds.length
+          );
+
+          if (nextIndex < 0) {
+            return;
+          }
+
+          const pageId = pageIds[nextIndex];
+
+          setFocusedPageId(pageId);
+
+          const page = pageRefs.current && pageRefs.current[pageId];
+
+          focusOnPage(page);
+
+          break;
+        }
+        default:
+          break;
+      }
+    },
+    [focusedPageId, isRTL, pageIds, pageRefs, gridRef, currentPageId]
+  );
+
+  // Rearrange pages
+  useKeyDownEffect(
+    ref,
+    { key: ['mod+up', 'mod+down', 'mod+left', 'mod+right'], shift: true },
```

Comment:
I agree with Brittany on constants and we could do the same for keys.

Reply:
Not opposed, but not sure we typically do this in the editor. Maybe something to bring up as a general code style question so things are consistent throughout the app?